server/controllers/userController.js

```diff
@@ -0,0 +1,320 @@
+    }
+
+    sendToken(user, 200, res);
+  } catch (error) {
+    return next(new ErrorHandler(error, 404));
+  }
+};
+
+//logout
+const logout = async (req, res, next) => {
+  res.cookie("token", null, {
+    expires: new Date(Date.now()),
+    httpOnly: true,
+  });
+  res.status(200).json({
+    success: true,
+    message: "Logged Out",
+  });
+};
+
+//for oneself
+const getUserDetails = async (req, res, next) => {
+  try {
+    const user = await User.findById(req.user.id);
+    res.status(200).json({
+      success: true,
+      user,
+    });
+  } catch (error) {
+    return next(new ErrorHandler(error.message, 404));
+  }
+};
+
+//get all users (for admin)
+const getAllUsers = async (req, res, next) => {
+  try {
+    const users = await User.find();
+    res.status(200).json({
+      success: true,
+      users,
+    });
+  } catch (error) {
+    return next(new ErrorHandler(error.message, 400));
+  }
+};
+
+//get a single user (for admin)
+const getSingleUser = async (req, res, next) => {
+  try {
+    const user = await User.findById(req.params.id);
+    if (!user) {
+      return next(
+        new ErrorHandler(`User Doesn't Exist with Id: ${req.params.id}`, 404)
+      );
+    }
+    res.status(200).json({
+      success: true,
+      user,
+    });
+  } catch (error) {
+    return next(new ErrorHandler(error.message, 400));
+  }
+};
+
+//update user role (for admin)
+const updateUserRole = async (req, res, next) => {
+  try {
+    const newUserData = {
+      name: req.body.name,
+      email: req.body.email,
+      role: req.body.role,
+    };
+
+    const user = await User.findByIdAndUpdate(req.user.id, newUserData, {
+      new: true,
+      runValidators: true,
+      useFindAndModify: false,
+    });
+
+    if (!user) {
+      return next(new ErrorHandler("User Does not Exist with this id ", 404));
+    }
+
+    res.status(200).json({
+      success: true,
+      user,
+    });
+  } catch (err) {
+    return next(new ErrorHandler(err.message, 500));
+  }
+};
+
+//delete user (for admin)
+const deleteUser = async (req, res, next) => {
+  try {
+    const user = await User.findByIdAndDelete(req.params.id);
+
+    if (!user) {
+      return next(new ErrorHandler("User Does not Exist with this id ", 404));
+    }
+
+    res.status(200).json({
+      success: true,
+      message: "User Deleted Successfully",
+    });
+  } catch (err) {
+    return next(new ErrorHandler(err.message, 500));
+  }
+};
+
+const forgotPassword = async (req, res, next) => {
+  try {
+    const user = await User.findOne({ email: req.body.email });
+    if (!user) {
+      return next(new ErrorHandler("User Not Found", 404));
+    }
+    const resetToken = await user.getResetPasswordToken();
+    await user.save({ validateBeforeSave: false });
+
+    const resetPasswordUrl = `http://localhost:5173/password/reset/${resetToken}`;
+
+    const message = `Your Reset Password Token Is ${resetPasswordUrl}`;

```

Comment:
use HTML markup code as the body of the email and shift that to a template folder or util.